math/include/mainframe/math/vector2.h

```diff
@@ -57,7 +57,7 @@ namespace mainframe {
 				return {std::abs(x), std::abs(y)};
 			}
 
-			VecType lerp(const VecType& other, NumberType timestep) const {
+			VecType lerp(const VecType& other, float timestep) const {
```

Comment:
this should be NumberType, if it's a vector of doubles, then this is invalid. NumberType is the template for it